### ISSUE TITLE
[DOXIA-685] Replace SinkEventAttributes#BOXED with #SOURCE to clearly separate between regular verbatim and verbatim source (code)

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-doxia#830]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** opened **[DOXIA-685](https://issues.apache.org/jira/browse/DOXIA-685?redirect=false)** and commented

Currently, verbatim text is inconsistently handled in the Sink API and formats apt, xhtml5, xdoc, fml and markdown.

`BOXED` implies that some both should be rendered in the output (see [APT spec](https://maven.apache.org/doxia/references/apt-format.html#verbatim-text) and its use for [xdoc source](https://maven.apache.org/doxia/references/xdoc-format.html#The_.3Csource.3E_tag)): since introduction of source code highlighting in skins, apt boxed more importantly indicates that this is source code to be likely highlighted.

In other words, from the API does not know this mix between boxed and source and about the look and feel at the end the output format and styling should decide rather than the sink.

For those reasons the following change is proposed:

* `BOXED` as original name will be deprecated and then removed w/o replacement (`SOURCE` is not really a replacement for the boxing)
* `Sink#verbatim()` and `Sink#verbatim(null)` will print verbatim text w/o any implications
* The `Xhtml5Sink` will make add to every verbatim block `<div class="verbatim">` instead of no class
* instead of `BOXED`, `SinkEventAttributes#SOURCE` to explicitly denotimate verbatim source code, `Xhtml5Sink` will add `<div class="verbatim source">`, the `Apt` format will turn `+--...--+` to verbatim source as well.
* The `EchoMacro` and `SnippetMacro` won't use `BOXED` anymore
* The `SnippetMacro` will have a `source=true` attribute which denotes source code, but only IF `verbatim=true`

#### input

* APT:

```
----
non-boxed verbatim text...
----
```

* Doxia Sink API:

```
Sink#verbatim(void);
Sink#text(String);
Sink#verbatim_(void);
```

* Markdown: TBD
* xdoc: TBD

#### output

* HTML

```
<div><pre>non-boxed verbatim text...</pre></div>
```

* HTML + CSS visual
  a CSS `border: 1px ...` is displayed in Fluido Skin (version?), that should not happen in a non-boxed case...
  rendered example on https://github.com/apache/maven-site/blob/master/content/apt/guides/introduction/introduction-to-the-lifecycle.apt
  !screenshot-1.png!

#### input

* APT:

```
+----+
boxed verbatim text...
+----+
```

* Doxia Sink API:

```
Sink#verbatim(SinkEventAttributeSet.BOXED);
Sink#text(String);
Sink#verbatim_(void);
```

* Markdown: TBD
* xdoc: TBD

#### output

* HTML

```
<div class="source"><pre>boxed verbatim text...</pre></div>
```

TBD: what is precisely done at Doxia level when rendering HTML, what is done at skin level server-side, what is done by JS/CSS at skin level browser time
* HTML + CSS visual

```
<div class="source"><pre class="prettyprint linenums">verbatim text...</pre></div>
```

HTML result is not focused on boxing or not boxing, but on source hightlighting and linenums (reason: this has been done over years, when introducing source code highlighting in Fluido, and discovering that what we needed in apt was more about activating/disabling highlighting, not so much about boxing. But this was never really specified anywhere, just explaining history 10 years after the fact)
rendered example on https://github.com/apache/maven-site/blob/master/content/apt/guides/introduction/introduction-to-the-lifecycle.apt
!screenshot-2.png!

#### input

```
Sink#verbatim(void);
Sink#text(String);
Sink#verbatim_(void);
```

#### output

!screenshot-3.png!

#### input

```
Sink#verbatim(SinkEventAttributeSet.SOURCE);
Sink#text(String);
Sink#verbatim_(void);
```

#### output

!screenshot-4.png!

---

**Affects:** 2.0.0-M4

**Attachments:**
- [screenshot-1.png](https://issues.apache.org/jira/secure/attachment/13054486/screenshot-1.png) (_204.39 kB_)
- [screenshot-2.png](https://issues.apache.org/jira/secure/attachment/13054487/screenshot-2.png) (_160.20 kB_)
- [screenshot-3.png](https://issues.apache.org/jira/secure/attachment/13054492/screenshot-3.png) (_80.57 kB_)
- [screenshot-4.png](https://issues.apache.org/jira/secure/attachment/13054493/screenshot-4.png) (_85.05 kB_)

**Issue Links:**
- [MSKINS-245](https://issues.apache.org/jira/browse/MSKINS-245) Maven Site 4 will break code highlight of site generated by Markdown
 (_**"causes"**_)
- [MSHARED-1182](https://issues.apache.org/jira/browse/MSHARED-1182) Drop incorrect usage of SinkEventAttributeSet#BOXED
 (_**"causes"**_)
- [MPIR-412](https://issues.apache.org/jira/browse/MPIR-412) Dependency report generates non-well-formed output if the POM of a depdendency cannot be parsed

- [DOXIA-762](https://issues.apache.org/jira/browse/DOXIA-762) Clarify code highlighting for verbatim text in Markdown and APT

- [DOXIA-731](https://issues.apache.org/jira/browse/DOXIA-731) Simplify HTML emitted from Sink.verbatim(...)

- [MSKINS-221](https://issues.apache.org/jira/browse/MSKINS-221) Update template to reflect verbatim block changes in Doxia 2.0.0-M6
 (_**"requires"**_)

**Remote Links:**
- [GitHub Pull Request #135
](https://github.com/apache/maven-doxia/pull/135)

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

If you really want to take BOXED and non-BOXED serious you need to modify (override) the `pre` CSS styles in bootstrap. I don't think that this is a good way.

**Comment by jira-importer** [comment]
**[Slawomir Jaranowski](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=slawekjaranowski)** commented

I don't expect that it will be so complicated :-)

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy), here is a snapshot of Maven Site with the fixes in Doxia und Fluido Skin: http://home.apache.org/~michaelo/maven/maven-site.tar.gz

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [73a00a9dead29b02c12c7d1365e2273ff502724a](https://gitbox.apache.org/repos/asf?p=maven-doxia.git;a=commit;h=73a00a9dead29b02c12c7d1365e2273ff502724a).
